blood/src/main/java/cz/cuni/mff/d3s/blood/utils/Result.java

```diff
@@ -68,4 +68,19 @@ public Success unwrap() {
     public Error unwrapError() {
         return expectError("unwrapError failed!");
     }
+
+    @FunctionalInterface
+    public static interface Function<T, R> extends java.util.function.Function<T, Result<R, Exception>> {
```

Comment:
Nice addition! That's really useful. I would just consider naming it more descriptively and adding a javadoc. It was not immediately clear to me what's it's purpose.

Reply:
> naming it more descriptively

What do you (as an experienced functional programmer) suggest?

> adding a javadoc

I will.

Reply:
> What do you (as an experienced functional programmer) suggest?

For example `CheckedFunction`. Not much more explanatory, but in general, naming classes with the same name as a class they inherit from is considered bad practice. It can be confusing when using those two classes at once...

And why do you make such snarky comment about my experiences? :disappointed: It's not even true and it's not really relevant in this case.